revpi_hat/hat.py

```diff
@@ -0,0 +1,122 @@
+from datetime import date
+import json
+from typing import List
+
+
+class RevPiHatEEPROMAttributeException(Exception):
+    pass
+
+
+class RevPiHatEEPROM:
+    def __init__(self, hat_eeprom_path: str = "/proc/device-tree/hat/", load_contents: bool = True) -> None:
+        self._hat_eeprom_path = hat_eeprom_path
+
+        self.uuid: str = None
+        self.format_version: int = None
+
+        self.vendor: str = None
+        self.product: str = None
+        self.product_id: int = None
+        self.product_id_revision: str = None
+        self.product_revision: int = None
+        self.product_version: str = None
+        self.product_version_major: int = None
+        self.product_version_minor: int = None
+
+        self.serial: int = None
+        self.eol_date: date = None
+        self.batch_number: int = None
+        self.first_mac_address: str = None
+
+        self._raw_values = {}
+
+        if load_contents:
+            self.load()
+
+    def load(self):
+        """
+        Load values from RevPi HAT EEPROM
+        """
+        self.uuid = self._hat_attribute("uuid")
+        self.format_version = self._hat_attribute_int("custom_0")
+
+        self.vendor = self._hat_attribute("vendor")
+        self.product = self._hat_attribute("product")
+        self.product_id = self._hat_attribute_int("product_id") + 100000
```

Comment:
A constant variable for "100000" to avoid the magic number?

Reply:
Makes sense. I'll update the PR and move the 100000 to a constant like `PRODUCT_ID_BASE=100000`